### ISSUE TITLE
Migration URL parsing is broken for versioned uberjars

### ISSUE THREAD

**Issue by jeaye** [igrishaev/pg2#14]
Hello, Ivan! I hope you're doing well. I just ran into a bug with pg2 and I'm open to making a PR but I wanted to check with you on the preferred behavior. The bug is with the regex used for parsing migration files:

```clojure
(def RE_FILE #"(?i).*?(\d+)\.(.*?)\.?(prev|next|up|down)\.sql$")
```

When this is used within a versioned uberjar, it parses incorrectly. For example, this is the URL for a migration file in my uberjar:

```
"jar:file:/home/jeaye/projects/bayaz/target/uberjar/bayaz-0.1.0-SNAPSHOT-standalone.jar!/migrations/015.create-nicks-from-joins.next.sql"
```

The `0.1.0-SNAPSHOT` is what trips up the regex here. The result is that every single migration file will be parsed with id `0`, which leads to duplicate migration file errors. (example in [regex101](https://regex101.com/r/bMePKe/1)) For a Clojure example:

```clojure
> (re-matches RE_FILE "jar:file:/home/jeaye/projects/bayaz/target/uberjar/bayaz-0.1.0-SNAPSHOT-standalone.jar!/migrations/015.create-nicks-from-joins.next.sql")

["jar:file:/home/jeaye/projects/bayaz/target/uberjar/bayaz-0.1.0-SNAPSHOT-standalone.jar!/migrations/015.create-nicks-from-joins.next.sql"
 "0"
 "1.0-SNAPSHOT-standalone.jar!/migrations/015.create-nicks-from-joins"
 "next"]

```

If we changed the regex to require a `/` before the number, we can fix this issue. That should work correctly for both uberjar and non-uberjar cases (since the migration dir will always be before the file name). However, it would change the behavior of the `parse-name` fn, if your users are calling it with a file name having no `/` before it. The resulting regex would just be:

```clojure
(def RE_FILE #"(?i).*?/(\d+)\.(.*?)\.?(prev|next|up|down)\.sql$")
```

Ultimately, I'm not sure how you'd prefer to handle this. For now, I've changed my jar's naming scheme, but I imagine this is going to bite a lot of people who're running versioned uberjars.

**Comment by igrishaev** [comment]
Hi! Thank you for reporting this. let me read through (it might take a while as I'm traveling with my family at the moment).

**Comment by igrishaev** [comment]
Dear @jeaye could you please review the PR? https://github.com/igrishaev/pg2/pull/15

**Comment by igrishaev** [comment]
Done, available in 0.1.16

**Comment by jeaye** [comment]
Thank you!
